src/OrchardCore.Modules/OrchardCore.Tenants/ViewModels/TenantViewModel.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Data;
+using OrchardCore.Environment.Shell;
+
+namespace OrchardCore.Tenants.ViewModels
+{
+    public class TenantViewModel : IValidatableObject
+    {
+        public string Description { get; set; }
+
+        [Required]
+        public string Name { get; set; }
+
+        public string DatabaseProvider { get; set; }
+
+        public string RequestUrlPrefix { get; set; }
+
+        public string RequestUrlHost { get; set; }
+
+        public string ConnectionString { get; set; }
+
+        public string TablePrefix { get; set; }
+
+        public string RecipeName { get; set; }
+
+        public string FeatureProfile { get; set; }
+
+        [BindNever]
+        public bool IsNewTenant { get; set; }
+
+        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
+            var shellHost = validationContext.GetService<IShellHost>();
+            var featureProfilesService = validationContext.GetService<IFeatureProfilesService>();
+            var databaseProviders = validationContext.GetServices<DatabaseProvider>();
+            var S = validationContext.GetService<IStringLocalizer<TenantViewModel>>();
+
+            var selectedProvider = databaseProviders.FirstOrDefault(x => x.Value == DatabaseProvider);
+
+            if (selectedProvider != null && selectedProvider.HasConnectionString && String.IsNullOrWhiteSpace(ConnectionString))
+            {
+                yield return new ValidationResult(S["The connection string is mandatory for this provider."], new[] { nameof(ConnectionString) });
+            }
+
+            if (String.IsNullOrWhiteSpace(Name))
+            {
+                yield return new ValidationResult(S["The tenant name is mandatory."], new[] { nameof(Name) });
+            }
+
+            if (!String.IsNullOrWhiteSpace(FeatureProfile))
+            {
+                var featureProfiles = featureProfilesService.GetFeatureProfilesAsync().Result;
```

Comment:
No `.Result` isn't going to fly. rest looks ok.

Reply:
Can I use `GetAwaiter().GetResult()` which is similar or you have better idea while the method is not `async`

Reply:
Nope, just as bad. the async stuff will not be able to go in IValidatableObject